bool sensor
int32 rack_num
